lib/hooks/core/hook-http.js

```diff
@@ -33,14 +34,12 @@ function requestWrap(request) {
       return request.apply(this, arguments);
     }
 
-    var localOptions = (typeof options === 'string') ?
-      url.parse(options) : options;
-
     // Don't trace ourselves lest we get into infinite loops
     // Note: this would not be a problem if we guarantee buffering
     // of trace api calls. If there is no buffering then each trace is
     // an http call which will get a trace which will be an http call
-    if (agent.isTraceAgentRequest(localOptions)) {
+    if (typeof options === 'object' &&
```

Comment:
This check seems unnecessary. `isTraceAgentRequest` checks that options is truthy, and has a headers property.